src/components/task-plan/reading.cjsx

```diff
@@ -82,91 +99,99 @@ SelectTopics = React.createClass
       </div>
     </BS.Modal>
 
+ReadingFooter = React.createClass
+  render: ->
+    classes = []
+    classes.push('disabled') unless @props.enabled
+    classes = classes.join(' ')
+
+    deleteLink = <a onClick={@props.onDelete}>Delete this plan</a> if @props.enabled
+
+    <span>
+      <BS.Button bsStyle="primary" className={classes} onClick={@props.onPublish}>Publish</BS.Button>
+      {deleteLink}
+    </span>
 
 ReadingPlan = React.createClass
   mixins: [Router.State, Router.Navigation]
 
   getInitialState: ->
     id = @getParams().id
-    if id
+    if (id)
       TaskPlanActions.load(id)
     else
       id = TaskPlanStore.freshLocalId()
-      TaskPlanActions.create(id, due_at: new Date())
+      plan = TaskPlanActions.create(id, due_at: new Date())
+
     {id}
 
   componentWillMount: -> TaskPlanStore.addChangeListener(@update)
   componentWillUnmount: -> TaskPlanStore.removeChangeListener(@update)
 
-  update: -> @setState {}
+  getPlanId: () ->
+    @getParams().id or @state.id
+
+  update: () ->
+    id = @getPlanId()
+    plan = TaskPlanStore.get(id)
+    if (id isnt plan.id)
+      @setState({id: plan.id})
+    else
+      @setState {}
 
   setDueAt: (value) ->
-    TaskPlanActions.updateDueAt(@state.id, value)
+    id = @getPlanId()
+    TaskPlanActions.updateDueAt(id, value)
 
   setTitle: ->
+    id = @getPlanId()
     value = @refs.title.getDOMNode().value
-    TaskPlanActions.updateTitle(@state.id, value)
+    TaskPlanActions.updateTitle(id, value)
 
-  publish: ->
-    {id} = TaskPlanStore.get(@state.id)
+  publishPlan: ->
+    id = @getPlanId()
     TaskPlanActions.publish(id)
     @transitionTo('editReading', {id})
 
-  renderTopics: (topicId) ->
-    topic = TocStore.getSectionInfo(topicId)
-    <li>{topic.number} - {topic.title}</li>
+  deletePlan: () ->
+    id = @getPlanId()
```

Comment:
Do we need a confirm for delete actions?  Is this really where the delete button for readings go?  If so, I can add it in.